client/foundries.go

```diff
@@ -682,10 +682,53 @@ func (a *Api) FactoryPatchConfig(factory string, cfg ConfigCreateRequest) error
 func (a *Api) FactoryListConfig(factory string) (*DeviceConfigList, error) {
 	url := a.serverUrl + "/ota/factories/" + factory + "/config/"
 	logrus.Debugf("FactoryListConfig with url: %s", url)
-	return a.DeviceListConfigCont(url)
+	return a.FactoryListConfigCont(url)
 }
 
 func (a *Api) FactoryListConfigCont(url string) (*DeviceConfigList, error) {
+	// A short cut as it behaves just the same
+	return a.DeviceListConfigCont(url)
+}
+
+func (a *Api) GroupCreateConfig(factory, group string, cfg ConfigCreateRequest) error {
+	data, err := json.Marshal(cfg)
+	if err != nil {
+		return err
+	}
+
+	url := a.serverUrl + "/ota/factories/" + factory + "/device-groups/" + group + "/config/"
+	logrus.Debug("Creating new device group config")
+	_, err = a.Post(url, data)
+	return err
+}
+
+func (a *Api) GroupDeleteConfig(factory, group, filename string) error {
+	url := a.serverUrl + "/ota/factories/" + factory + "/device-groups/" + group + "/config/" + filename + "/"
+	logrus.Debugf("Deleting config file: %s", url)
+	_, err := a.Delete(url, nil)
+	return err
+}
+
+func (a *Api) GroupPatchConfig(factory, group string, cfg ConfigCreateRequest) error {
+	data, err := json.Marshal(cfg)
+	if err != nil {
+		return err
+	}
+
+	url := a.serverUrl + "/ota/factories/" + factory + "/device-groups/" + group + "/config/"
```

Comment:
Don't we need a filename of a config group that is being patched?

Reply:
A patch can apply to entire config which has many filenames